### PR TITLE
Cache included liquid templates

### PR DESCRIPTION
**Pull request by hishamco**
Address issue #36

### CODE REVIEW

**Comment by hishamco** [comment]
I can ride off a lot of code by introducing:
```csharp
private class FluidViewTemplate : BaseFluidTemplate<FluidViewTemplate>
{

}
```
/cc @sebastienros

File: Fluid/Ast/IncludeStatement.cs [hunk 1]
```diff
@@ -56,38 +60,56 @@ public IncludeStatement(Expression path, Expression with = null, IList<AssignSta
...
 
             return Completion.Normal;
         }
 
+        private static IFluidTemplate ParseTemplate(string path, IFileProvider fileProvider, TemplateContext context)
+        {
+            return _memoryCache.GetOrCreate(path, entry =>
+            {
+                var fileInfo = fileProvider.GetFileInfo(path);
+                entry.SlidingExpiration = TimeSpan.FromHours(1);
```
Comment (sebastienros) [inline Fluid/Ast/IncludeStatement.cs]: Static value.

Comment (hishamco) [reply Fluid/Ast/IncludeStatement.cs]: We can introduce an option to control this for both included templates and other liquid templates if it's needed

Comment (sebastienros) [reply Fluid/Ast/IncludeStatement.cs]: We could but we need to enforce there is one for security reasons. If you can make it an option too then good.

Comment (hishamco) [reply Fluid/Ast/IncludeStatement.cs]: Ok

**Comment by hishamco** [comment]
The `CacheOption` for `FluidMvcViewEngine` is straightforward, I don't wanna include tag rely on the `FluidMvcViewEngine` options, coz may someone use is Fluid without `FluidMvcViewEngine`

Any idea @sebastienros ?

**Comment by sebastienros** [comment]
Then don't make it an option.

**Comment by hishamco** [comment]
So can I simplify the parsing using:
```csharp
private class FluidViewTemplate : BaseFluidTemplate<FluidViewTemplate>
{

}
```
or leave it as its

**Comment by sebastienros** [comment]
I don't understand the relationship with this PR and FluidViewTemplate already inherits from it. Why make it private then?

**Comment by hishamco** [comment]
Just an accessor, I wanna call `FluidViewTemplate.TryParse` instead of creating a new parser instance, like [this](https://github.com/sebastienros/fluid/blob/dev/Fluid/Ast/IncludeStatement.cs#L91-L115)

**Comment by sebastienros** [comment]
`CreateParser` is cheap, use it, and remove the `else` cases: https://github.com/sebastienros/fluid/blob/dev/Fluid/Ast/IncludeStatement.cs#L97

**Comment by hishamco** [comment]
I can't remove it because it will not compile, anyhow we can use the available `CreateParser()` without introducing something else

**Comment by sebastienros** [comment]
> I can't remove it because it will not compile 

If I remove it and the tests pass will you trust me ?

**Comment by hishamco** [comment]
If is that what's the need to relay on FluidParserFactoryKey?

**Comment by hishamco** [comment]
Hope you will like the last change that I made

**Comment by sebastienros** [comment]
If Include can be used outside of FluidMvc (which I assume is the case) then the template can be of any type (as it's customizable) hence you can use a specific type, and need to use whatever is passed in the ambient values. And not use this class anywhere in the statements.

**Pull request closed without merging**